store/git/git.py

```diff
@@ -0,0 +1,128 @@
+gflags.DEFINE_string('git_root', None, 'The absolute path to local git root')
+gflags.DEFINE_string('method', 'get', 'FileSystemStore method either get, put')
+gflags.DEFINE_string('status_file', None, 'The status file to record success or failure of the operation')
+gflags.DEFINE_string('suppress_error', False, 'Suppress Error')
+
+gflags.MarkFlagAsRequired('key')
+gflags.MarkFlagAsRequired('store_location')
+
+
+FLAGS = gflags.FLAGS
+
+class LocalGitStore(object):
+  """A class to get and put file in local GIT File System"""
+
+  class LocalGitStoreError(Exception):
+    def __init__(self, message):
+      self.message = message
+
+    def __str__(self):
+      return 'Method failed due to {1}'.format(self.method, self.message)
+
+
+  def __init__(self, store_location, key, method, git_root, suppress_error, status_file=None):
+    try:
+      self.git_root = git_root or os.environ['GIT_ROOT']
+      self.store_location = os.path.join(self.git_root, store_location)
+      self.key = key
+      self.method = method
+      self.status_file = status_file
+      self.suppress_error = suppress_error
+    except KeyError:
+      raise LocalGitStore.LocalGitStoreError("Git root not found. Either use --git_rot or  bazel command line flag --action_env=GIT_ROOT=`pwd`")
+
+  def __enter__(self):
+    self.status_code = 0
+    return self
+
+  def __exit__(self, t, v, traceback):
+    if self.status_file:
+      with open(self.status_file, "w") as f:
+        f.write("{0}".format(self.status_code))
+
+  @contextmanager
+  def _execute(self, commands):
+    try:
+     for command in commands:
+       self.status_code = subprocess.check_call(command)
+    except subprocess.CalledProcessError as e:
+       self.status_code = e.returncode
+       if self.suppress_error:
+          print(e)
+          return
+       raise LocalGitStore.LocalGitStoreError(e)
+
+
+  def get(self, get):
+    """Get file and copy it to bazel workspace"""
+    file_location = os.path.join(self.store_location, self.key)
+    self._execute(commands = [
+       ['cp', file_location, os.path.join(self.git_root, get)]
+    ])
+
+  def put_if_not_exists(self, src):
+    file_location = os.path.join(self.store_location, self.key)
+    if not os.path.exists(file_location):
+      self.put(src)
+
+  def put(self, src):
+    """Put file from Bazel workspace to local file system"""
+    file_location = os.path.join(self.store_location, self.key)
+    self._execute(commands = [
+      ['mkdir', '-p', os.path.dirname(file_location)],
+      ['cp', src, file_location]
+    ])
+     
+
+def main(unused_argv):
+  with LocalGitStore(store_location=FLAGS.store_location,
+                     method=FLAGS.method,
+                     key=FLAGS.key,
+                     git_root = FLAGS.git_root,
+                     suppress_error=FLAGS.suppress_error,
+                     status_file=FLAGS.status_file) as git_store:
+    if FLAGS.method == "get":
```

Comment:
nit, style: could you put flags validation outside the context manager?

Reply:
done